purpose.tex

```diff
@@ -5,13 +5,34 @@ \section{Purpose}
 Laboratory protocols are critical to biological research and development, but can be difficult to communicate or reproduce due to the differences in context, skills, and resources between different projects, investigators, and organizations.
 The Protocol Activity Markup Language (PAML) aims to address this problem by providing a data model for description of laboratory protocols that is unambiguous enough for precise interpretation and automation, yet simultaneously abstract enough to support reuse and adaptation.
 
+It behoves us to unpack the notion of replicability further, in order to better undestand the requirements on PAML as an artifact.  What does it mean to be ``unambiguous enough for precise interpretation and automation'' and `` support reuse and adaptation''? More specifically, what \emph{computational tasks} must PAML support?  We list here a bare minimum set of tasks; there are opportunities to add many more:
+
+\begin{description}
+\item[Instantiation] An \emph{abstract} protocol specifies a \emph{recipe} for scientific investigation.  A concrete \emph{instance of} a protocol is an execution of the recipe at a specific time and place, on a specific set of equipment, etc.  This implies at least two representational requirements:
+  \begin{description}
+  \item[Instantiation Proper] We must be able to create some persistent data structure that represents the protocol instance.
+  \item[Derivation] We must be able to represent the relationship between the protocol instance and the recipe from which it is derived.
+  \item[Modification] We must be able to record modifications made to the original recipe, for example, to enable it to be executed at a lab with different equipment from the lab at which it was originated, to enable the protocol to be scaled up or scaled down, etc.
```

Comment:
I'm not sure this is something that we want to record under Instantiation --- I think this is a matter of making a derived work and recording the relationship.  If we want support execution of modified protocols, that goes against the decision to disallow dynamical patching of protocols that is recorded in https://github.com/SD2E/paml/issues/15

A use pattern that I think may end up being common is to have a protocol be "too strict" to be instantiated in a lab, and thus need modification. Rather than the modification being a "patch", however, it would be a generalization that allows it to be instantiated in both prior labs and the new labs, which would be released as an updated version of the existing protocol. The instantiation would then not be a modification, but an instantiation of a later version than first intended.

Reply:
I have captured that in a new version that I will push momentarily, incorporating your example.

One issue about instantiation is what we think instantiation should be. Is instantiation very much like making a new version of the protocol (so relatively complete)? Or do we have in mind something where the instance is a different sort of object, more lightweight, that simply records a link between a block of parameters (OPIL, presumably), the protocol (PAML), and possibly some additional metadata and annotation?  Either seems possible, but we should probably state a rationale for choosing one or the other.

Reply:
@rpgoldman I expect that we'll have both: as you note, the lightweight is OPIL, the heavy-weight is an execution trace for PAML.